packages/langium/src/dependency-injection.ts

```diff
@@ -76,14 +76,22 @@ const __requested__ = Symbol();
  */
 function _resolve<I, T>(obj: any, prop: string | symbol | number, module: Module<I, T>, injector: I): T[keyof T] | undefined {
     if (prop in obj) {
+        if (obj[prop] instanceof Error) {
+            throw new Error('Construction failure. Please make sure that your dependencies are constructable.', {cause: obj[prop]});
+        }
         if (obj[prop] === __requested__) {
             throw new Error('Cycle detected. Please make "' + String(prop) + '" lazy. See https://langium.org/docs/di/cyclic-dependencies');
         }
         return obj[prop];
     } else if (prop in module) {
         const value: Module<I, T[keyof T]> | ((injector: I) => T[keyof T]) = module[prop as keyof T];
         obj[prop] = __requested__;
-        obj[prop] = (typeof value === 'function') ? value(injector) : _inject(value, injector);
+        try {
+            obj[prop] = (typeof value === 'function') ? value(injector) : _inject(value, injector);
+        } catch (error) {
+            obj[prop] = error;
```

Comment:
Please add an `instanceof Error` check here as well (theoretically you can throw anything in JS).

Reply:
ok